src/client/runner/utils/get-line-rect-intersection.js

```diff
@@ -0,0 +1,62 @@
+function getLineYByXCoord (startLinePoint, endLinePoint, x) {
+    if (endLinePoint.x - startLinePoint.x === 0)
+        return null;
+
+    return Math.round(startLinePoint.y +
+                      (x * (endLinePoint.y - startLinePoint.y) + startLinePoint.x * (startLinePoint.y - endLinePoint.y)) / (endLinePoint.x - startLinePoint.x));
+}
```

Comment:
Formatting is still extremely weird. Maybe refactor subexpressions in to vars to make this look better?